### ISSUE TITLE
Get the TestGrid UI deployed with equivalent functionality

### ISSUE THREAD

**Issue by michelle192837** [kubernetes-sigs/testgrid#4]
tl;dr: Listing the work remaining to get the TestGrid UI MVP (this repo) to have equivalent functionality to the existing closed-source UI (at https://testgrid.k8s.io/).

(Needs the import to be done first, ref #2).

The UI is basically a collection of web components that read data from the [TestGrid API](https://github.com/GoogleCloudPlatform/testgrid/tree/main/cmd/api#api) and handle specific functions within the UI. The main ones are navigation (to move between pages), the index (list of dashboard groups and dashboards), summary (display recorded summary data), and the grid component (the main logic; displays recorded test history in an interactive grid).

Most of this should require some familiarity with (or willingness to learn) frontend tech (TypeScript, web components, Lit, CSS, etc.). This is not currently deployed anywhere, but once the import is complete you should be able to run it locally.

It may help to have familiarity with (or willingness to learn) Golang as well so you can work with the API.

Note: This list is a work-in-progress and is expected to update with missing items as we come across them.
All the work below should come with sufficient testing (unit and manual tests).

- [ ] Navigation component
  - [ ] (Is there any remaining work?)
- [ ] Index page
  - [ ] (Is there any remaining work?)
- [ ] Summary component
  - [ ] (Is there any remaining work?)
- [ ] Grid component
  - [ ] Update component for performance (should handle 1 million elements easily)
  - [ ] Ensure all statuses (color, icon, message) are handled
  - [ ] Properly handle grouped/nested rows
  - [ ] Properly handle all column and row headers (e.g. frozen to top and left sides, and scroll appropriately)
  - [ ] Handle right-click context menu on cells (e.g. can file a bug and open test results)
  - [ ] Add code-search link on failing rows
  - [ ] Add tooltips (on cells and column headers) 
- [ ] Accessibility: Verify that components are accessible
- [ ] Style: Unify styles across all components / pages
- [ ] Documentation: Update documentation for all components
- [ ] Deploy
  - [ ] Deploy UI in a community-owned project (coordinate with K8s Infra)
  - [ ] Hook UI up to existing TestGrid API endpoint
  - [ ] Add link for new UI (separate from testgrid.k8s.io)
  - [ ] If all is well, switch testgrid.k8s.io to point to the new UI

**Comment by manzil-infinity180** [comment]
I would like to help with this pending checkbox to fill it with a tick mark!

**Comment by benjaminapetersen** [comment]
/retitle New open-source UI checklist for feature parity with old closed-source UI

**Comment by michelle192837** [comment]
Ah sorry, I totally forgot some context for how links work.

So in general links for the tab are customizable via LinkTemplate fields in the configuration. (See https://github.com/GoogleCloudPlatform/testgrid/blob/main/pb/config/config.proto#L523 for the proto message, used for fields like https://github.com/GoogleCloudPlatform/testgrid/blob/main/pb/config/config.proto#L569).
The way this currently works (in closed-source UI) is:
- the UI takes a relevant LinkTemplate (e.g. for opening the link on a cell, `open_test_template`)
- Parse LinkTemplate and substitute any parameters (e.g. `<test-id>` pulls the cell ID)
  - https://github.com/GoogleCloudPlatform/testgrid/blob/497f7c85e651610c13d4e688ee35ee049880a040/util/links.go is not battle-tested, but it has a gist of the kinds of values that are substituted for link templates

There is an issue I missed here, which is that the API doesn't serve the LinkTemplates for a tab at the moment. I think the relevant issue in GCP/testgrid is https://github.com/GoogleCloudPlatform/testgrid/issues/1200. (I roughly know how to add this, I just haven't had time to do so).

If you want to check what the configuration for a current tab is in testgrid.k8s.io btw, you can go to "About" > "View Configuration" on a given tab to get the config for both the tab and its backing test group. ^^ (The endpoint backing this page is NOT on the TestGrid API so we can't use it; we need to add a new endpoint instead).
  - For instance for https://testgrid.k8s.io/sig-testing-testgrid#test-testgrid-npm (config at https://testgrid.k8s.io/q/config/sig-testing-testgrid/test-testgrid-npm):

```
open_test_template:  {
  url:  "https://prow.k8s.io/view/gs/<gcs_prefix>/<changelist>"
}
```

- Note that `gcs_prefix` comes from the configuration. `changelist` is better known as a `build number` and comes from the `build number` column header data.

My suggestions:
- If you want to make progress on the right-click context menu, you can handle link substitution later, and just work with the existing link from the web component for now.
- As a second step, you can add basic link substitution for the values mentioned in [links.go](https://github.com/GoogleCloudPlatform/testgrid/blob/497f7c85e651610c13d4e688ee35ee049880a040/util/links.go) that can be fetched from the grid.
- Then later when the API is updated, we can add the remaining stuff for 1) getting the right link template from configuration, and 2) substituting any config-only values (like `gcs_prefix`).

**Comment by michelle192837** [comment]
Yup, that sounds correct to me.
